Initial Sheets API implementation

talvarez0507 opened the pull request:
This PR includes the basic authorization features of the sheets API and the creation of a blank spreadsheet. Additionally, it creates some buttons on the UI for logging in, exporting (the blank file), and logging out.

talvarez0507 committed:
implement basic sheets functionality

talvarez0507 committed:
link sheets JS file and add buttons

hollyyuqizheng commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,133 @@
+
+// Authorization scopes required by the API; multiple scopes can be
+// included, separated by spaces.
+var SCOPES = 'https://www.googleapis.com/auth/spreadsheets';
```
I personally added what the scope means in the variable name in my implementation, so maybe something like SCOPE_READ_WRITE would be more clear in indicating what this scope lets the user to do.

talvarez0507 replied:
Done.

hollyyuqizheng commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,133 @@
+ */
+function handleError(error) {
+  $('#sheets-message').removeClass('d-none');
+  if (error.error === 'popup_closed_by_user') {
```
Maybe it's better if these error codes are extracted into constant global variables.

talvarez0507 replied:
Done.

hollyyuqizheng reviewed (approved):
Some suggestions on variable naming.

hollyyuqizheng commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,133 @@
+  $('#sheets-message').show();
+}
+
+function handleExportSchedule() {
```
Adding some comments here would help along side the TODO items.

talvarez0507 replied:
Done.

talvarez0507 committed:
fix incorrect file name

talvarez0507 committed:
update scope variable name

talvarez0507 committed:
extract error codes to variables

talvarez0507 committed:
separate error handling functions

talvarez0507 committed:
add comments for explanation

raulcruise commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,146 @@
+// Client ID and API key from the Developer Console
+var CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+var API_KEY = 'AIzaSyAcS8-F743Xy2ZTSYxxZIB0nY5kDQQ8i7E';
```
I don't think our API Key should be public.

talvarez0507 replied:
Done.

talvarez0507 committed:
remove API key

jamesnam-google commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,146 @@
+// Authorization scopes required by the API; multiple scopes can be
+// included, separated by spaces.
+var SCOPE_READ_WRITE = 'https://www.googleapis.com/auth/spreadsheets';
+// Error codes that the process can throw
```
Consider using object for different error codes.

var ERROR_CODE = {
  ...: ...
  ...: ...
};

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,146 @@
+  gapi.auth2.getAuthInstance().signOut();
+}
+
+
```
Unnecessary line break

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,146 @@
+ * message on the UI.
+ */
+function handleAuthorizationError(error) {
+  $('#sheets-message').removeClass('d-none');
```
Consider assigning $('#sheets-message') to a variable

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/sheetsApi.js:
```diff
@@ -0,0 +1,146 @@
+ * message on the UI.
+ */
+function handleExportError(reason) {
+  $('#sheets-message').removeClass('d-none');
```
Same here. And I think you can chain these.

var $sheetsMessage = $('...');
$sheetsMessage.removeClass(...)
    .text(...)
    .show();

talvarez0507 replied:
Done.

jamesnam-google reviewed (approved):
Approving the change with comments.

talvarez0507 committed:
implement change requests

put error codes into a single object, remove extra blank line, extract $('#sheets-message') into a variable for readability
